python/ql/test/library-tests/frameworks/serverless/aws_lambda/function/lambda_function.py

```diff
@@ -0,0 +1,2 @@
+def lambda_handler(event, context): # $ remoteFlow=event
+    return "OK"
```

Comment:
Usually tests of remote-flow-sources use the inline taint-test ([here's the one for flask](https://github.com/github/codeql/blob/36bdadfc36a31cca7cb759760a81c89183204bcb/python/ql/test/library-tests/frameworks/flask/taint_test.py)), so it would be nice if we could do that here too, instead of introducing a new type of test.

I would also like to see _use_ of the event in these tests, since that's really what is important.

I have sketched out the test body below, but obviously requires deletion and introduction of some ql test files :blush: 

```suggestion
def lambda_handler(event, context):
    ensure_tainted(
        event, # $ tainted
        # event is usually a dict, see https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html
        event["key"], # $ tainted
        event["key"]["key2"], # $ tainted
        event["key"][0], # $ tainted
        # but can also be a list
        event[0], # $ tainted
    )
    return "OK"
```

Reply:
> Usually tests of remote-flow-sources use the inline taint-test ([here's the one for flask](https://github.com/github/codeql/blob/36bdadfc36a31cca7cb759760a81c89183204bcb/python/ql/test/library-tests/frameworks/flask/taint_test.py)), so it would be nice if we could do that here too, instead of introducing a new type of test.

That is nicer indeed, so I did that, thanks. Looking at the flask test, I noted the routed parameter. Should we really model the server-less functions as `Server.RequestHandler`s?

Reply:
that's a really good point. It would make a lot of sense if these were modeled as RequestHandlers :+1: